### PR TITLE
Add custom errors to docs

### LINKED ISSUE #4477: Include custom errors in documentation
### Description

After OpenZeppelin Contracts 5.0 and with the addition of custom errors, there are some use cases of listing the custom errors in the documentation. We should consider adding them to the docs in a (maybe) similar way to how we show events docs.

### PR DESCRIPTION
**Pull request by ernestognw**
This PR adds custom error documentation to the docs templates. Also, it adds EIP-6093 to the interfaces documentation given the amount of references to it. Most of the error links would be broken otherwise.

Fixes #4477 

#### PR Checklist

- [ ] Tests
- [x] Documentation
- [ ] Changeset entry (run `npx changeset add`)

### CODE REVIEW

File: contracts/interfaces/draft-IERC6093.sol [hunk 1]
```diff
@@ -4,7 +4,7 @@ pragma solidity ^0.8.19;
 /**
  * @dev Standard ERC20 Errors
  * Interface of the ERC6093 custom errors for ERC20 tokens
- * as defined in https://eips.ethereum.org/EIPS/eip-6093
+ * as defined in https://eips.ethereum.org/EIPS/eip-6093[the EIP]
```
Comment (Amxx) [inline contracts/interfaces/draft-IERC6093.sol:7]: I'd write it as 
```
* Interface of the https://eips.ethereum.org/EIPS/eip-6093[ERC-6093] custom errors for ERC20 tokens
```

Comment (ernestognw) [reply contracts/interfaces/draft-IERC6093.sol]: I agree, extending it was too repetitive, and didn't notice. Updated

**Comment by Amxx** [comment]
This looks good to me if we want to add the errors in the interface page, next to the other interface functions... but maybe we should have a specific, separate, section. WDYT ?

**Comment by ernestognw** [comment]
> This looks good to me if we want to add the errors in the interface page, next to the other interface functions... but maybe we should have a specific, separate, section. WDYT ?

Ideally, I'd like to give the same importance to all interfaces listed and this alternative seems a bit unbalanced:

Captura de pantalla 2023-07-25 a la(s) 15 04 03


What's your idea for having a specific place for them?

**Comment by frangio** [comment]
In `AccessControl` we should mention the error that is emitted, as discussed in https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4475. Can we add it in this PR?

**Review by frangio — APPROVED** [review]
Looks nice. I'm fine with the current structure of the interfaces page. We could probably do better than this but I don't have ideas atm and I wouldn't block merging on that.

**Comment by ernestognw** [comment]
> In `AccessControl` we should mention the error that is emitted, as discussed in #4475. Can we add it in this PR?

Of course!

**Review by frangio — APPROVED** [review]

**Pull request merged**